thoth/lab/underscore.py

```diff
@@ -30,7 +30,102 @@ class _Underscore(object):
     def __init__(self, df: pd.DataFrame):
         self._df: pd.DataFrame = df
 
-    def str_join(self, cols: typing.List[str] = None, *, sep: str = ""):
+    def get(self, key: str, *, default=DEFAULT, **kwargs) -> pd.Series:
+        """Get record from column entries by given key."""
+        attrs = key.split(".", 1)
+
+        if len(attrs) == 2:
+            col, attrs = attrs
+            values = self._df[col]._.get(attrs, default=default, **kwargs)
+        else:
+            col, = attrs
+            values = self._df[col]
+
+        return values
+
+    def flatten(
+        self,
+        col: str,
+        record_paths: typing.Union[list, dict] = None,
+        *,
+        columns: list = None,
+        drop: bool = False,
+        inplace: bool = False,
+        default: typing.Union[str, dict] = None,
+        **kwargs,
+    ) -> pd.DataFrame:
+        """Flatten specific column of dictionaries or lists by extracting records from each entry.
+        
+        If the column contains dictionaries, they will be flatten into columns. If the column
+        contains lists, they will be flatten into rows.
+
+        A column can not contain combination of lists and dictionaries.
+        """
+        record_paths = record_paths or set()
+        gather_record_paths = not record_paths
+
+        # validate
+        dtype = None
+        for idx, entry in self._df[col].iteritems():
+            entry_type = type(entry)
+
+            if pd.isna(dtype):
+                dtype = entry_type
+            elif entry_type is not dtype:
+                raise TypeError(f"Multiple types found in column '{col}': {[dtype, entry_type]}")
+
+            if gather_record_paths and dtype is dict:
+                record_paths.update(entry.keys())
+
+        if dtype is None:
+            return self._df
+
+        if isinstance(dtype(), typing.Mapping):
+            df_flat = self._df[col]._.flatten(record_paths, columns=columns, **kwargs)
+
+        elif isinstance(dtype(), (list, set, tuple)):
```

Comment:
`issubclass` not to call constructor?

Reply:
Ah nice, thanks! I thought that something like that must've existed :D